scale/src/test/java/com/strategyobject/substrateclient/scale/readers/BooleanArrayReaderTest.java

```diff
@@ -0,0 +1,35 @@
+package com.strategyobject.substrateclient.scale.readers;
+
+import com.strategyobject.substrateclient.common.utils.HexConverter;
+import lombok.val;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+class BooleanArrayReaderTest {
+
+    @Test
+    void read() throws IOException {
```

Comment:
I think it might be a problem since arrays of primitive types don't have a base type except `Object`. So `assertArrayEquals` has overloads for different types that has to be known at compile time to do a proper dispatch.
 I can do some runtime dispatch with `instanceof` and casts, but it won't be any better.

Reply:
I can overcome this problem by wrapping an assertion in a `ThrowingConsumer`. Looks like this:
```
new TestCase<>(new BooleanArrayReader(), "0x18010101000001", x -> assertArrayEquals(new boolean[]{true, true, true, false, false, true}, (boolean[]) x))
```
looks a little ugly though.

Reply:
I would propose something like this:

Reply:
Another option - you can resolve asserter from map by type.

Reply:
yes, might work. I'll give it a try.